docs/dev/uap/permissions.rst

```diff
@@ -0,0 +1,90 @@
+Permissions
+===========
+
+Permissions are divided into data access permissions (DAPs) and device permissions. Permissions are checked on a user
+object using the default django.contrib.auth has_perm user-model methods, which take a permission string and an
+optional object for row-level permissions.
+
+Objects which foreign key onto users (a.k.a User-related objects) can, in a limited way, define row-level permissions.
+In reality, these permissions aren't decided row-by-row for efficiency, but instead define the following class method
+to return an iterable of objects for which a requesting user has permissions given the related user:
```

Comment:
Maybe it should be a QuerySet only? Just to enforce somewhat reasonable ways of using the permission system on row-level?